woocommerce/woocommerce-ios
# [Woo POS] Add totals view. Shuffle views on checkout flow

Partially addresses #12763
Closes: #12764

This PR adds a initial `Totals` view, and shuffles the order of the different views depending if we're `building` or `finalizing` checkout. Most likely the view style will change, so I haven't taken much time with the UI details (eg: "Take payment" most likely will be removed or not be a button)

## Description

![Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2024-05-22 at 15 54 04](https://github.com/woocommerce/woocommerce-ios/assets/3812076/eb4b9df2-0dc4-42f1-92cd-e2876114823b)

## Changes
- New `TotalsView`. At the moment this just renders a list of products in-cart, and three buttons at the bottom for payment collection
- Added an `OrderStage` enum that indicates where in the process are we, `building` or `finalizing`.

## Testing instructions
- Add a bunch of products to the cart
- Tap `Checkout`
- Observe how the views shuffle and the Cart appears to the left while a new Totals view appears to the right, with a list of all products added to the cart.
- Remove products, observe that cart state is updated
- Tap `Add More`
- Observe we're back to order `building` and cart state is updated
